sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1486,13 +1486,18 @@ __ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, cos, cos)
       return esimd::saturate<T>(simd<T, N>(res_raw));                          \
   }                                                                            \
                                                                                \
-  /** Scalar version.                                                       */ \
+  /** (vector, scalar) version.                                             */ \
```

Comment:
Shouldn't (scalar, vector) variant be supported as well?

Reply:
This is actually a good point, thanks.

Reply:
It turns out, scalar vector is supported as well, as implicit conversion works for the first parameter. I updated the test adding this (and few other similar) test cases.